Fix a small parsing bug with Context.Attrs

kkaneda opened the pull request:
It used ":" as a separator while a comment in flags.go says Attrs is a comma-separated list.

kenjikaneda committed:
Fix comments regarding the separator of --attrs and --stores

- --attrs is a colon-separated list, not a comma-separated list
- Fix example values of --stores, which had "|" as a separator

Also add a test on Context.Attrs parsing

spencerkimball commented:
LGTM. Thanks for addressing this.

bdarnell commented on server/context.go:
```diff
@@ -165,12 +166,12 @@ func (ctx *Context) initEngine(attrsStr, path string) (engine.Engine, error) {
 // filtering empty strings (i.e. ",," will yield no attributes.
 // Returns the list of strings as Attributes.
-func parseAttributes(attrsStr string) proto.Attributes {
+func parseAttributes(attrsStr string, sep string) proto.Attributes {
```
Do we need to be able to parse attributes in two different ways? Can we just use the same separator everywhere or is there some reason why we want `:` in some places and `,` in others?

kkaneda replied:
Hmm, we can use `:` instead of `,` as separator of --attrs, but I thought that it's a bit uncommon convention. 

We cannot simply change the separator of --stores since the flag uses both `:` and `,` as its separator. The flag description says "specify a comma-separated list of stores, specified by a colon-separated list of device attributes ...".

spencerkimball replied:
Well commas separate the stores and colon separates multiple attributes within a store.

For example:

-store=hdd:7200rpm=/mnt/hda1,ssd:gpu=/mnt/ssd01

Would mean one store with attributes {hdd, 7200rpm} at /mnt/hda1 and another store with attributes {ssd, gpu} at /mnt/ssd01.

bdarnell replied:
So then it sounds to me like the node-level attributes should be parsed with `:` instead of `,`. `--store=hdd:7200rpm=/mnt/hda1,ssd=/mnt/ssd1 --attrs=us-east:gpu` (so no code change, just update the docs of --attrs)

spencerkimball replied:
Yeah, that's a reasonable change.

kkaneda replied:
Sounds good to me too. Will do.

kkaneda commented:
Updated the PR. I also fixed comments in server/cli/start.go that had "|" as a separator. @bdarnell , @spencerkimball could you take another look?

Thanks,
-Kenji

spencerkimball commented:
LGTM. Thanks for adding a test.
